pageserver/src/remote_storage/storage_sync.rs

```diff
@@ -413,19 +427,28 @@ fn storage_sync_loop<
 ) -> anyhow::Result<()> {
     let remote_assets = Arc::new((storage, RwLock::new(index)));
     while !crate::tenant_mgr::shutdown_requested() {
```

Comment:
What's also not needed, IMO, is another shutdown signal in `walreceiver.rs`, the `interrupt_receiver` one.
IMO, we can use one global shutdown hook in all places, but I'll check that separately.

Reply:
IMO it is not enough to use one global shutdown hook, it should be per tenant at least, and  more granular shutdown hooks for every component may also be needed in case we want some specific component shutdown ordering. Another option is to have global hook for global services, ones that are shared between tenants. E.g. page service listener or http endpoint. I think the best place to discuss this is #988